File: src/main/java/com/alluvialtrading/tools/BenchmarkGenerator.java
package com.alluvialtrading.tools;

import com.alluvialtrading.data.Quote;
//import com.alluvialtrading.database.ASXBackTestingLib;
import com.alluvialtrading.vbroker.BackTestingLib;
import com.alluvialtrading.vbroker.LSEBackTestingLib;

import java.util.Date;

public class BenchmarkGenerator {

	private BackTestingLib connector = null;
	private TraderLib lib = new TraderLib();
	
	public static void main(String[] args) {
		new BenchmarkGenerator();
	}
	
	public BenchmarkGenerator() {
		connector = new LSEBackTestingLib();
		Date start = connector.getFirstTradeDate();
		Date end = connector.getLastTradeDate();
		String currentDate = lib.dateToISODateString(start);
		
		String endString = lib.dateToISODateString(end);
		Date date = null;
		while (!currentDate.equals(endString)) {
			//
			//System.out.println(currentDate);
			String[] allSecurities = connector.getAllSecurities(currentDate);
			for (String security : allSecurities) {
				security = security.trim();
				date = lib.convertISODateTimeString(currentDate);
				Date endOfDay = lib.combineDateTime(date, "23:59:59.000");
				double open = connector.getOpenPrice(security, date);
				double close = connector.getClosePrice(security, date);
				if (security.equals("BHP")) {
					System.out.println("*" + currentDate + " " + security);
				}
				Quote precloseQuote = connector.getQuote(security, currentDate, connector.getCloseTime());
				double preclose = 0;
				if (null != precloseQuote) {
					preclose = precloseQuote.getMid();
				}
				double high = connector.getMaxPrice(security, date, endOfDay);
				double low = connector.getMinPrice(security, date, endOfDay);
				int tcount = connector.getTcount(security, date, endOfDay);
				int volume = connector.getVolume(security, date, endOfDay, 0);
				double value = connector.getValue(security, date, endOfDay, 0);
				double minSpread = connector.getMinSpread(security, currentDate, currentDate, false);
				System.out.println(currentDate + " " + security);
				connector.insertBenchmark(currentDate, security, open, close, preclose, high, low, tcount, volume, value, minSpread);
				
			}
			currentDate = connector.getTradingDate(currentDate, 1);
		}
		
	}
}

File: src/main/scala/com/alluvia/tools/BenchmarkGenerator.scala
package com.alluvia.tools

import java.util.Date

import com.alluvia.markets._
import com.alluvia.database.BackTestingLib
import com.alluvialtrading.data.Quote
import com.alluvialtrading.tools.TraderLib

object BenchmarkGenerator {
  def main(args: Array[String]) {
    new BenchmarkGenerator with LSE
  }
}


abstract class BenchmarkGenerator extends BackTestingLib with Market {

  val library = new TraderLib
  var start: Date = library.combineDateTime("2011-04-01", "00:00:01.000")
  var end: Date = getLastTradeDate
  var currentDate: String = library.dateToISODateString(start)
  var endString: String = library.dateToISODateString(end)
  var date: Date = null
  while (!(currentDate == endString)) {
    val allSecurities: Array[String] = getAllSecurities(getMarketName, currentDate)
    for (rawSecurity <- allSecurities) {
      val security = rawSecurity.trim
      date = library.convertISODateTimeString(currentDate)
      val endOfDay: Date = library.combineDateTime(date, "23:59:59.000")
      val open: Double = getOpenPrice(getMarketName, security, date)
      val close: Double = getClosePrice(getMarketName, security, date)
      val precloseQuote: Quote = getQuote(getMarketName, security, currentDate, getCloseTime)
      val preclose: Double =  if (null != precloseQuote) precloseQuote.getMid else 0.0

      val high: Double = getMaxPrice(getMarketName, security, date, endOfDay)
      val low: Double = getMinPrice(getMarketName, security, date, endOfDay)
      val tcount: Int = getTcount(getMarketName, security, date, endOfDay)
      val volume: Int = getVolume(getMarketName, security, date, endOfDay, 0)
      val value: Double = getValue(getMarketName, security, date, endOfDay, 0)
      val minSpread: Double = getMinSpread(getMarketName, security, currentDate, currentDate, false)
      System.out.println(currentDate + " " + security)
      insertBenchmark(getMarketName, currentDate, security, open, close, preclose, high, low, tcount, volume, value, minSpread)
    }
    currentDate = getTradingDate(getMarketName, currentDate, 1)
  }

}

